File: spec/functional/inside_government/format_success_spec.rb
require "functional/test_helper"

describe "Inside Gov Format success graph" do
  include StubApiFromFixtures

  describe "svg rendering" do
    it "should be visible" do
      visit "/performance/dashboard/government"
      wait_until do
        all("#inside-gov-format-success *[name()='svg']").count == 1
      end

      page.find("#inside-gov-format-success *[name()='svg']").visible?.should be_true
    end

  end
end

File: spec/javascripts/specs/modules/inside-government/FormatSuccessSpec.js
describe("inside gov format success graph", function () {

    var stubGraphContainer = $("<div id='inside-gov-format-success'></div>");

    var STUB_DATA = {
        "response_info":{
            "status":"ok"
        },
        "id":"http://datainsight-frontend.dev.gov.uk/performance/dashboard/government/format-success.json",
        "details":{
            "source":["Totally made up"],
            "data":[
                {
                    "format":"small grey",
                    "entries":5000,
                    "percentage_of_success":50
                }
            ]
        },
        "updated_at":"2012-10-30T09:27:34+00:00"
    };

    
    // outdated, replace with new inside-gov specific test
    xit("should insert an svg onto the page", function () {
        stubGraphContainer.appendTo('body');

        GOVUK.Insights.InsideGovernment.formatSuccess(STUB_DATA);

        expect($("#inside-gov-format-success").length).toBeGreaterThan(0);
        
        stubGraphContainer.remove();
    });

});